internal/users/tempentries/tempentries.go

```diff
@@ -61,7 +61,7 @@ func (r *TemporaryRecords) UserByName(name string) (types.UserEntry, error) {
 //
 // Returns the generated UID and a cleanup function that should be called to remove the temporary user once the user was
 // added to the database.
-func (r *TemporaryRecords) RegisterUser(name string) (uid uint32, cleanup func() error, err error) {
+func (r *TemporaryRecords) RegisterUser(name string) (uid uint32, cleanup func(), err error) {
```

Comment:
Do we have a reason to have named values here (a part from clarity, that I agree is nice)?

As the cleanup can be a bit dangerous, since it's going to be set to `nil` by default, and calling it by mistake before being set, will lead to a crash without being caught before.

Reply:
> Do we have a reason to have named values here (a part from clarity, that I agree is nice)?

Just for clarity, which I find important though.

> As the cleanup can be a bit dangerous, since it's going to be set to nil by default, and calling it by mistake before being set, will lead to a crash without being caught before.

True. IMO we should be careful not to do that. I would prefer to keep the named values to make it clear from the function signature what is being returned.

Reply:
In case that can also be handled via the comment though.

I've not a  strong opinion, nor it's a blocker, but I just would like that at least the linter would block us if returning without setting it...

Reply:
Where do you want to add a comment?

Reply:
I mean using function comments instead of named return values.

Reply:
But those are not visible in the autocompletion list in the IDE (in contrast to the function signature)

Reply:
I guess it depends on the IDE; vscode shows them... However again not a blocker, but a potential danger to consider.